keanu-project/src/test/java/io/improbable/keanu/algorithms/variational/KLDivergenceTest.java

```diff
@@ -0,0 +1,31 @@
+package io.improbable.keanu.algorithms.variational;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.algorithms.mcmc.MetropolisHastings;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.vertices.ProposalVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.binary.AdditionVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+
+import java.util.Collections;
+
+public class KLDivergenceTest {
+    @Test
+    public void testKLDivergence() {
+        GaussianVertex v1 = new GaussianVertex(0., 1.);
+        ConstantDoubleVertex v2 = new ConstantDoubleVertex(0.1);
+        AdditionVertex v3 = new AdditionVertex(v1, v2);
+
+        BayesianNetwork network = new BayesianNetwork(v3.getConnectedGraph());
+        NetworkSamples samples = MetropolisHastings
+            .withDefaultConfig()
+            .getPosteriorSamples(network, Collections.singletonList(v1), 1000);
+
+        ProposalVertex q = new GaussianVertex(0.1, 1.);
+
+        assertTrue(KLDivergence.compute(q, samples) < 3.0);
```

Comment:
This assertion is a bit arbitrary. Do we know what the KLDiv for these two distributions should be?

Reply:
Maybe you can calculate the KLDiv for Gaussian(0.1, 1.0) and Gaussian(3.0, 1.0) and assert that the first is less than the second?

Reply:
Theoretically, the divergence should be 0. But we are sampling p here, so it'll just be arbitrarily low.

I like the idea of comparing the divergences between two different Gaussians!

Reply:
Although, I see potential for flaky tests ... But I think we will be OK if the two qs are _very_ different.